### ISSUE TITLE
Logging Issues

### ISSUE THREAD

**Issue by jwoosnam** [Psychoanalytic-Electronic-Publishing/OpenPubArchive-Content-Server#68]
@nrshapiro I have a couple of issues:
- Reports - I have managed to logon using https://stage-api.pep-web.rocks/docs#/, but I can't then run a report "Could not validate credentials"

- How do I find the doc Author from Metadata?

Could we arrange to have a chat?  I have a meeting with David on Wednesday....

**Comment by nrshapiro** [comment]
@jwoosnam 

- Reports - I have managed to logon using https://stage-api.pep-web.rocks/docs#/, but I can't then run a report "Could not validate credentials"

Report access depends on the user being an admin, but I was out of town last Thursday - Sunday and I have yet to implement PaDS-based admin access to the API using PaDS login (and of course I removed the OPAS based authentication which was the way to have an account with admin access before).  [I saw your message about reviewing your proposed PEPSecure API changes, but I haven't had a chance yet, given I just returned last night.]

The document authors are listed in the documentListItem for any document returned as authorMast, and in detail in the XML data in documentInfoXML.  A list of documentListItems is returned for a number of operations, including search.

- How do I find the doc Author from Metadata?

Search for "philanthropy" and in responseSet you get a list of dicts (documentListItems in the model), for example this item:
```
  {
        "documentID": "IJP.046.0396A",
        "docType": "REV",
        "documentRef": "Hood, J. (1965) Child Psychotherapy: Edited by Mary R. Haworth.  (New York and London:  Basic Books, 1964.  Pp. 459.  $8.50.  50s.). International Journal of Psychoanalysis  46:396",
        "documentRefHTML": "<p class=\"citeas\"><span class=\"authors\">Hood, J.</span> (<span class=\"year\">1965</span>) <span class=\"title\">Child Psychotherapy: Edited by Mary R. Haworth.  (New York and London:  Basic Books, 1964.  Pp. 459.  $8.50.  50s.)</span>. <span class=\"sourcetitle\">International Journal of Psychoanalysis </span> <span class=\"pgrg\">46</span>:<span class=\"pgrg\">396</span></p>",
        "documentInfoXML": "<artinfo arttype=\"REV\" artsource=\"A1V2\" j=\"IJP\" artsubtype=\"BKR\" id=\"IJP.046.0396A\" ISSN=\"0020-7578\"><artyear>1965</artyear>\n<artvol>46</artvol>\n<artpgrg style=\"arabic\" prefixused=\"\">396</artpgrg>\n<artvar>A</artvar><arttitle>Child Psychotherapy</arttitle>\n<artsub>Edited by Mary R. Haworth.  (New York and London:  Basic Books, 1964.  Pp. 459.  $8.50.  50s.)</artsub><artauth hidden=\"false\"><aut role=\"reviewer\" alias=\"false\" listed=\"true\" asis=\"false\" authindexid=\"Hood, James\"><nfirst type=\"FIRST\">James</nfirst><nlast>Hood</nlast></aut>\n</artauth>\n</artinfo>\n",
        "title": "<arttitle>Child Psychotherapy</arttitle>",
        "authorMast": "James Hood",
        "PEPCode": "IJP",
        "sourceTitle": "International Journal of Psychoanalysis ",
        "sourceType": "journal",
        "kwicList": [
          "intellect, as Barzun (1959)1 has pertinently described them, the cult of art, and <span class='searchhit'>philanthropy</span>, and we would agree wholeheartedly with him that\nThough the",
          "genius of Freud was unswervingly intellectual it has given birth to a large progeny of adaptors, who, from generous as well as selfish motives, put <span class='searchhit'>philanthropy</span> first.  They mean to cure or at least to"
        ],
        "kwic": "intellect, as Barzun (1959)1 has pertinently described them, the cult of art, and <span class='searchhit'>philanthropy</span>, and we would agree wholeheartedly with him that\nThough the . . . genius of Freud was unswervingly intellectual it has given birth to a large progeny of adaptors, who, from generous as well as selfish motives, put <span class='searchhit'>philanthropy</span> first.  They mean to cure or at least to",
        "vol": "46",
        "year": "1965",
        "lang": "en",
        "issn": "0020-7578",
        "doi": null,
        "issue": null,
        "pgRg": "396",
        "pgStart": "396",
        "pgEnd": null,
        "docLevel": 1,
        "updated": "2020-09-17T07:05:55+00:00",
        "score": 5.1576333,
        "rank": 1,
        "stat": null,
        "accessClassification": "archive",
        "accessLimited": true,
        "accessLimitedReason": "This is a summary excerpt from the full text of the document. The full content of the document is available to subscribers. ",
        "accessLimitedDescription": "The full content of the document is available to subscribers. ",
        "accessLimitedPubLink": null
      },

```
If you were logged in and requested the document itself, you'd get:

```
 [
      {
        "documentID": "IJP.046.0396A",
        "docType": "REV",
        "documentRef": "Hood, J. (1965) Child Psychotherapy: Edited by Mary R. Haworth.  (New York and London:  Basic Books, 1964.  Pp. 459.  $8.50.  50s.). International Journal of Psychoanalysis  46:396",
        "documentRefHTML": "<p class=\"citeas\"><span class=\"authors\">Hood, J.</span> (<span class=\"year\">1965</span>) <span class=\"title\">Child Psychotherapy: Edited by Mary R. Haworth.  (New York and London:  Basic Books, 1964.  Pp. 459.  $8.50.  50s.)</span>. <span class=\"sourcetitle\">International Journal of Psychoanalysis </span> <span class=\"pgrg\">46</span>:<span class=\"pgrg\">396</span></p>",
        "documentInfoXML": "<artinfo arttype=\"REV\" artsource=\"A1V2\" j=\"IJP\" artsubtype=\"BKR\" id=\"IJP.046.0396A\" ISSN=\"0020-7578\"><artyear>1965</artyear>\n<artvol>46</artvol>\n<artpgrg style=\"arabic\" prefixused=\"\">396</artpgrg>\n<artvar>A</artvar><arttitle>Child Psychotherapy</arttitle>\n<artsub>Edited by Mary R. Haworth.  (New York and London:  Basic Books, 1964.  Pp. 459.  $8.50.  50s.)</artsub><artauth hidden=\"false\"><aut role=\"reviewer\" alias=\"false\" listed=\"true\" asis=\"false\" authindexid=\"Hood, James\"><nfirst type=\"FIRST\">James</nfirst><nlast>Hood</nlast></aut>\n</artauth>\n</artinfo>\n",
        "title": "<arttitle>Child Psychotherapy</arttitle>",
        "authorMast": "James Hood",
        "PEPCode": "IJP",
        "sourceTitle": "International Journal of Psychoanalysis ",
        "sourceType": "journal",
        "kwicList": [],
        "kwic": "",
        "vol": "46",
        "year": "1965",
        "lang": "en",
        "issn": "0020-7578",
        "doi": null,
        "issue": null,
        "pgRg": "396",
        "pgStart": "396",
        "pgEnd": null,
        "abstract": "\n                    <p class=\"heading\"><span class='pub_year'>(1965)</span>. <span class='source_title'>International Journal of Psychoanalysis , </span><span class='vol'>(46)</span><span class='issue'></span><span class='pgrg'>:396</span></p>\n                    <p class=\"title\"><arttitle>Child Psychotherapy</arttitle></p>\n                    <p class=\"title_author\">James Hood</p>\n                    <div class=\"abstract\"><!DOCTYPE html>\n<html xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<head>\n<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\">\n<title class=\"head title\"></title>\n<link rel=\"stylesheet\" type=\"text/css\" href=\"pepepub.css\">\n</head>\n<div id=\"abs\" class=\"abs\">\n<p class=\"para\" data-pagehelper2=\"firstchild\" id=\"idm45925922075704\">This handsomely produced volume tries to bring together a vast collection of diverse excerpts from already published material.  The attempt of its editor to make available a 'comprehensive and representative presentation of theoretical and methodological approaches' is unfortunately foiled by its all-embracing character.  ...       ",
        "document": "<!DOCTYPE html>\r<html xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<head>\n<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\">\n<title class=\"head title\">James Hood: Child Psychotherapy</title>\n<link rel=\"stylesheet\" type=\"text/css\" href=\"pepepub.css\">\n</head>\n<body><div class=\"pepkbd3\" data-ver=\"2012-08-13.FULL\">\n<div id=\"front\" class=\"frontmatter\">\n<p class=\"banner\"><a class=\"anchor\" name=\"IJP.046.0396A\" id=\"IJP.046.0396A\"></a><a class=\"toc-link\" href=\"/#/ArticleList/?journal=IJP\"><img src=\"https://stage-api.pep-web.rocks/v2/Documents/Image/bannerIJPLogo.gif\" alt=\"Journal Logo\"></a></p>\n<div class=\"pubinfotop\"><span class='pub_year'>(1965)</span>. <span class='source_title'>International Journal of Psychoanalysis , </span><span class='vol'>(46)</span><span class='issue'></span><span class='pgrg'>:396</span></div>\n ..."
        "docPagingInfo": {
          "page": null,
          "page_limit": null,
          "page_offset": 0
        },
        "docLevel": 1,
        "updated": "2020-09-17T07:05:55+00:00",
        "score": 1,
        "rank": 1,
        "stat": null,
        "term": "SearchHits(*:*)",
        "termCount": 0,
        "accessClassification": "archive",
        "accessLimited": false,
        "accessLimitedReason": "This archive content is available for you to access",
        "accessLimitedDescription": "This archive content is available for you to access",
        "accessLimitedPubLink": null
      }
    ]
```

**Comment by jwoosnam** [comment]
Reporting: See attachment
<img width="1379" alt="Curl" src="https://user-images.githubusercontent.com/70103034/96463453-a4e93d80-121e-11eb-8dc1-cfb085af065f.png">

Author:
My interface that collects the data uses /v2/Metadata/Contents/{SourceCode}/{SourceVolume}/.  

When I try https://stage-api.pep-web.rocks/v2/Metadata/Contents/IJP/101/?year=%2A&limit=200 it returns the below.  authorMast is Null??:

    "responseSet": [
      {
        "documentID": "IJP.101.0479A",
        "documentRef": "Marks, L. (2019) Driven to re-member: André Green’s use of the negative in the work of Berlinde de Bruyckere. International Journal of Psychoanalysis  101:479-495",
        "documentRefHTML": "<p class=\"citeas\"><span class=\"authors\">Marks, L.</span> (<span class=\"year\">2019</span>) <span class=\"title\">Driven to re-member: André Green’s use of the negative in the work of Berlinde de Bruyckere</span>. <span class=\"sourcetitle\">International Journal of Psychoanalysis </span> <span class=\"pgrg\">101</span>:<span class=\"pgrg\">479-495</span></p>",
        "documentInfoXML": "<artinfo arttype=\"ART\" j=\"IJP\" doi=\"10.1080/00207578.2019.1683848\" ISSN=\"0020-7578\" id=\"IJP.101.0479A\">\n    <artyear>2019</artyear>\n    <artvol>101</artvol>\n    <artiss>3</artiss>\n    \n    <artpgrg style=\"arabic\" prefixused=\"\">479-495</artpgrg>\n    <arttitle>Driven to re-member: Andr&#233; Green&#8217;s use of the negative in the work of Berlinde de Bruyckere</arttitle>\n    <artauth hidden=\"false\"><aut listed=\"true\" authindexid=\"Marks, Lesley\" role=\"author\" affid=\"AF1\" alias=\"false\" asis=\"false\"><nfirst type=\"ALLGIVEN\" initials=\"None\">Lesley</nfirst><nlast>Marks</nlast><nbio>\nCONTACT Lesley Marks <webx type=\"email\" url=\"mailto:lesleymou@gmail.com\">lesleymou@gmail.com</webx> Yehoshua Bin Nun Street 10/7, Tel Aviv 62643, Israel\n</nbio></aut>\n\t\t            \n<autaff affid=\"AF1\">\n                    \n\n<addr>\n<ln>Independent Researcher, Tel Aviv</ln>\n<ln>Israel</ln>\n</addr>\n                    \n</autaff>\n            </artauth>\n<artkwds><impx type=\"KEYWORD\">art</impx><impx type=\"KEYWORD\">applied psychoanalysis</impx><impx type=\"KEYWORD\">French psychoanalysis</impx><impx type=\"KEYWORD\">body</impx><impx type=\"KEYWORD\">psychic pain</impx></artkwds>\n\n</artinfo>\n",
        "authorMast": null,
        "PEPCode": "IJP",
        "vol": "101",
        "year": "2019",
        "pgRg": "479-495",
        "pgStart": "479",
        "pgEnd": "495",
        "score": 2
      },

**Comment by nrshapiro** [comment]
Oh--that's one of the earlier endpoints and it now has a bug given some schema changes.  The field used to populate that in the endpoint doesn't exist any more (or was renamed).  I'm looking into it.  Good catch!

**Comment by jwoosnam** [comment]
OK - I will leave my code as is and it should pick up the author once you have a chance to fix the problem.

**Comment by nrshapiro** [comment]
@jwoosnam 

- I verified that the Logging endpoint only requires a api_key.  
- Fixed the bug as of ver 2020.1019.1 RE the authorMast field in the /v2/Metadata/Contents endpoints.
